lightningd/jsonrpc.c

```diff
@@ -380,6 +380,16 @@ void json_add_address(struct json_result *response, const char *fieldname,
 		json_add_string(response, "address", addrstr);
 		json_add_num(response, "port", addr->port);
 	}
+	else if (addr->type == ADDR_TYPE_TOR_V2) {
+		json_add_string(response, "type", "torv2");
+		json_add_string(response, "address", tal_fmt(addr,"%s.onion",b32_encode(addrstr, (u8 *)addr->addr,2)));
```

Comment:
Why not just use `fmt_wireaddr_without_port`?  Then you can keep `b32_encode` `static` inside `common/tor.c`. (this is also a good opportunity to refactor similar code for IPv4 and IPv6 above)